utils/test/incident_analysis_tests/ClusterAnalysisTestCase.py

```diff
@@ -29,70 +29,81 @@
+
+        UnitTestUtilities.checkGeoObjects([Configuration.incidentInputGDB, \
+                                           self.incidentScratchGDB, \
+                                           self.toolboxUnderTest, \
+                                           self.inputPointsFeatures])
+
     def tearDown(self):
         if Configuration.DEBUG == True: print(".....ClusterAnalysisTestCase.tearDown")
-        UnitTestUtilities.deleteScratch(Configuration.incidentScratchGDB)
-
-    def test_cluster_analysis_pro(self):
-        '''test_cluster_analysis_pro'''
-        if Configuration.DEBUG == True: print(".....ClusterAnalysisTestCase.test_cluster_analysis_pro")
-        arcpy.ImportToolbox(Configuration.patterns_ProToolboxPath, "iaTools")
-        outputClusterFeatures = os.path.join(Configuration.incidentScratchGDB, "outputClusters")
-        runToolMessage = "Running tool (Cluster Analysis - Pro)"
-        arcpy.AddMessage(runToolMessage)
-        Configuration.Logger.info(runToolMessage)
-        try:
-            arcpy.ClusterAnalysis_iaTools(self.inputPointsFeatures, outputClusterFeatures)
-        except:
-            msg = arcpy.GetMessages(2)
-            self.fail('Exception in ClusterAnalysis_iaTools for Pro toolbox \n' + msg)
-        clusterCount = int(arcpy.GetCount_management(outputClusterFeatures).getOutput(0))
-        self.assertEqual(clusterCount, int(37))
-
-    def test_cluster_analysis_desktop(self):
-        '''test_cluster_analysis_desktop'''
-        if Configuration.DEBUG == True: print(".....ClusterAnalysisTestCase.test_cluster_analysis_desktop")
-        arcpy.ImportToolbox(Configuration.patterns_DesktopToolboxPath, "iaTools")
-        outputClusterFeatures = os.path.join(Configuration.incidentScratchGDB, "outputClusters")
-        runToolMessage = "Running tool (Cluster Analysis - Desktop)"
+        UnitTestUtilities.deleteScratch(self.incidentScratchGDB)
+
+    def test_cluster_analysis(self):
+        '''test_cluster_analysis'''
+        if Configuration.DEBUG == True: print(".....ClusterAnalysisTestCase.test_cluster_analysis")
+        arcpy.ImportToolbox(self.toolboxUnderTest, self.toolboxUnderTestAlias)
+        outputClusterFeatures = os.path.join(self.incidentScratchGDB, "outputClusters")
+        runToolMessage = "Running tool (Cluster Analysis)"
         arcpy.AddMessage(runToolMessage)
         Configuration.Logger.info(runToolMessage)
         try:
             arcpy.ClusterAnalysis_iaTools(self.inputPointsFeatures, outputClusterFeatures)
         except:
             msg = arcpy.GetMessages(2)
-            self.fail('Exception in ClusterAnalysis_iaTools for Desktop toolbox \n' + msg )
+            self.fail('Exception in ClusterAnalysis_iaTools toolbox \n' + msg)
         clusterCount = int(arcpy.GetCount_management(outputClusterFeatures).getOutput(0))
-        self.assertEqual(clusterCount, int(37))
+        self.assertGreater(clusterCount, int(10))
```

Comment:
Here I got a great idea for testing feature output from unit tests from Dev Summit. Build a custom Assert based on arcpy.FeatureCompare results. The *arcpyAssert.py* mixin was a complete copy from Chris Moravec: https://github.com/MoravecLabs/ArcpyTestExample. Kind of makes it easier to test equivalency between feature classes, rather than faking it by comparing row counts, spatial references, field names, etc.